examples/async_clock/src/clock.rs

```diff
@@ -0,0 +1,53 @@
+use std::time::Duration;
+
+use chrono::Local;
+use gloo_timers::future::sleep;
+use tokio_stream::wrappers::ReceiverStream;
+use wasm_bindgen_futures::spawn_local;
+
+/// Simple Clock
+pub struct Clock;
+
+const ONE_SEC: Duration = Duration::from_secs(1);
+
+impl Clock {
+    /// Create a new clock.
+    pub fn new() -> Self {
+        Clock
+    }
+
+    /// Returns a stream of time updates.
+    pub fn stream_time(&self) -> ReceiverStream<String> {
+        // Create a sender and receiver pair.
+        let (tx, rx) = tokio::sync::mpsc::channel(10);
+
+        // Spawn a background task that will send the current time to the receiver every second.
+        spawn_local(async move {
+            // Wait a bit before starting the stream, so the user can observe the "initialized"
+            // state.
+            sleep(Duration::from_secs(2)).await;
+
+            loop {
+                let formatted_time = Self::nice_now();
+                tx.send(formatted_time).await.expect("Failed to send time");
+                sleep(ONE_SEC).await;
+            }
+        });
+
+        // Return the receiver. Note that this method returns immediately, and that any work is done
+        // in the background.
+        ReceiverStream::new(rx)
+    }
+
+    fn nice_now() -> String {
+        let current_time = Local::now();
+        current_time.to_rfc2822()
+    }
+}
```

Comment:
I understand what you mean. The purpose of the example, however, is demonstrate how asynchronous tasks can be integrated with yew components. Your suggestion is without arguing more effective for the specific task, but it is possibly less useful to demonstrate the integration of async work in general.

We use yew for our UI, mainly because of our Rust skills writing backend microservices. The use of channels is (at least for us) a common pattern to stream work-items between threads. 

Others may be looking for similar ways to implement these patterns in the UI too.

[Tokio 1.21.0](https://github.com/tokio-rs/tokio/releases/tag/tokio-1.21.0) now intentionally supports WASM, so I'm hopeful that the dependency quirks will be better supported in future releases.  

I've added a comment to the code to explain that there are more efficient ways to accomplish the specific task.

Reply:
That said... I do get your comment about tokio, and less is more. Reworked the example to use `futures::channel` and drop the tokio dependency completely.

Reply:
Good feedback. Let me see if I can work these various options in the example.

Reply:
I've reworked the examples to showcase the various async methods you've mentioned. There is a bit more going on, but it does demonstrate the different ways people can use futures depending on their use case.